### DISCUSSION TITLE
How to resolve "Variable not allowed in type expression" errors?

### DISCUSSION THREAD

**Discussion by rohan-mehta** [strawberry-graphql/strawberry#3486] [General]
Hi, I have code like this:
```
MyGraphQLEnum = strawberry.enum(MyEnum, name="MyGraphQLEnum")

def do_something(data: MyGraphQLEnum):
  pass
```

On the `do_something` line, I get a type checking error from Pyright:
```
Variable not allowed in type expression
(variable) MyGraphQLEnum: Unknown
```

Just wondering if anyone has suggestions on how to fix this?

**Comment by patrick91** [comment]
@rohan-mehta unfortunately now, we'd need to change how we create enums to properly type it I think 🤔

**Reply by rohan-mehta** [reply]
@patrick91 makes sense, thanks for responding!
